Firestore/core/src/core/field_filter.cc

```diff
@@ -78,40 +82,31 @@ const char* CanonicalName(Filter::Operator op) {
 
 }  // namespace
 
-FieldFilter FieldFilter::Create(FieldPath path,
+FieldFilter FieldFilter::Create(const FieldPath& path,
                                 Operator op,
-                                FieldValue value_rhs) {
+                                google_firestore_v1_Value value_rhs) {
   if (path.IsKeyFieldPath()) {
     if (op == Filter::Operator::In) {
-      return KeyFieldInFilter(std::move(path), std::move(value_rhs));
+      return KeyFieldInFilter(path, value_rhs);
     } else if (op == Filter::Operator::NotIn) {
-      return KeyFieldNotInFilter(std::move(path), std::move(value_rhs));
+      return KeyFieldNotInFilter(path, value_rhs);
     } else {
-      HARD_ASSERT(value_rhs.type() == FieldValue::Type::Reference,
-                  "Comparing on key, but filter value not a Reference.");
       HARD_ASSERT(!IsArrayOperator(op),
                   "%s queries don't make sense on document keys.",
                   CanonicalName(op));
-      return KeyFieldFilter(std::move(path), op, std::move(value_rhs));
+      return KeyFieldFilter(path, op, value_rhs);
     }
   } else if (op == Operator::ArrayContains) {
-    return ArrayContainsFilter(std::move(path), std::move(value_rhs));
+    return ArrayContainsFilter(path, value_rhs);
 
   } else if (op == Operator::In) {
-    HARD_ASSERT(value_rhs.type() == FieldValue::Type::Array,
-                "IN filter has invalid value: %s", value_rhs.type());
-    return InFilter(std::move(path), std::move(value_rhs));
+    return InFilter(path, value_rhs);
   } else if (op == Operator::ArrayContainsAny) {
-    HARD_ASSERT(value_rhs.type() == FieldValue::Type::Array,
-                "arrayContainsAny filter has invalid value: %s",
-                value_rhs.type());
-    return ArrayContainsAnyFilter(std::move(path), std::move(value_rhs));
+    return ArrayContainsAnyFilter(path, value_rhs);
   } else if (op == Operator::NotIn) {
-    HARD_ASSERT(value_rhs.type() == FieldValue::Type::Array,
-                "notIn filter has invalid value: %s", value_rhs.type());
-    return NotInFilter(std::move(path), std::move(value_rhs));
+    return NotInFilter(path, value_rhs);
   } else {
-    Rep filter(std::move(path), op, std::move(value_rhs));
+    Rep filter(std::move(path), op, value_rhs);
```

Comment:
Should `path` be moved, since it's received by the method as a const reference? This looks on the surface to be a missed occurrence of changing `std::move(path)` to simply `path`.